Makefile

```diff
@@ -16,48 +13,27 @@ GOBIN=$(shell go env GOBIN)
 endif
 
 .PHONY: all
-all: manager test
+all: prepare test
 
 .PHONY: run
 run: manifests ## Run component against the configured Kubernetes cluster in ~/.kube/config
 	go run ./cmd/managers/$(COMPONENT)/main.go
 
-.PHONY: install
-install: manifests ## Install component CRDs into a cluster
-	kustomize build config/$(COMPONENT)/crd | kubectl apply -f -
-
-.PHONY: deploy
-deploy: manifests ## Deploy component controller in the configured Kubernetes cluster in ~/.kube/config
-	cd config/$(COMPONENT)/manager && kustomize edit set image controller=${IMG}
-	kustomize build config/$(COMPONENT)/default | kubectl apply -f -
-
-.PHONY: docker-build ## Build component docker image
-docker-build: test
-	docker build . --build-arg component=$(COMPONENT) -t ${IMG}
-
-.PHONY: docker-push
-docker-push: docker-build ## Push the docker image
-	docker push ${IMG}
-
 .PHONY: test
 test: generate fmt vet manifests ## Run tests
 	go test ./... -coverprofile cover.out
 
-.PHONY: package
-package: generate fmt vet manifests ## Package for distribution
+.PHONY: compile
```

Comment:
What's the point of this target? It publishes images (with forced -L = local) and their coordinates are lost.

I need a way to publish images to a possibly non local, configurable registry (ie use KO_DOCKER_REPO) and apply the resulting yaml

Reply:
> I need a way to publish images to a possibly non local, configurable registry

That's exactly why the command shouldn't be a make target. Using `ko` directly is more expressive and powerful. What you're looking for is:

```
ko apply -f config/riff-${component}.yaml
```

The intent of the compile target is to make sure that the containers can be built, we don't do anything with those containers. It's only really used as part of the CI test job, I'd be ok dropping the target from the makefile and only running `ko resolve` from the CI workflow.